frontend/src/components/BreadcrumbNavigation.js

```diff
@@ -5,64 +5,70 @@ import { useNavigation } from './NavigationContext';
 import {fetchAncestors, fetchRegion} from '../api';
 
 const BreadcrumbNavigation = () => {
-    const { selectedRegion, setSelectedRegion, selectedHierarchy } = useNavigation();
-    const [breadcrumbItems, setBreadcrumbItems] = useState([{ id: null, name: 'World', hasSubregions: true }]);
+  const { selectedRegion, setSelectedRegion, selectedHierarchy } = useNavigation();
+  const [breadcrumbItems, setBreadcrumbItems] = useState([{ id: null, name: 'World', hasSubregions: true }]);
+  const [error, setError] = useState(null);
```

Comment:
I would suggest using a global state, aka Provider, for the errors. It will make it possible to set an error from any component and render it in a unified way.
For example, take a look here:
https://github.com/OhmSpectator/track-your-regions/blob/30a0f4b13dffe76be146e15be4128322197c7b18/frontend/src/components/NavigationContext.js#L10